crates/sui-core/src/checkpoints/mod.rs

```diff
@@ -94,6 +94,56 @@ pub struct PendingCheckpoint {
     pub details: PendingCheckpointInfo,
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum PendingCheckpointV2 {
+    // This is an enum for future upgradability, though at the moment there is only one variant.
+    V2(PendingCheckpointV2Contents),
+}
```

Comment:
This is purely a consensus message right? as in this doesn't have anything to do with the public facing CheckpointContents that is synced in state-sync correct?

Reply:
Correct

Reply:
I don't really love `Contents` in the name either but needed something to disambiguate the enum name from the struct name, other ideas welcome

Reply:
This is not a consensus message, its stored in the epoch store

Reply:
Sorry you're right Mark, I meant "correct" that it has nothing to do with public facing state sync messages, but it is not a consensus message